### PR TITLE
SP-7877 ott UI testing

### PR DESCRIPTION
**Pull request by Nevazhnovu**

### CODE REVIEW

File: ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift [hunk 1]
```diff
@@ -198,7 +198,11 @@ class SPNativeLongButton: SPNativeUI {
             onText = try container.decode(String.self, forKey: .onText)
             offText = try container.decode(String.self, forKey: .offText)
             customText = try container.decode(String.self, forKey: .customText)
-            text = try container.decode(String.self, forKey: .text)
+            do {
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:201]: Why is this necessary?

File: Example/Podfile [hunk 2]
```diff
@@ -15,13 +15,13 @@ end
 target 'ConsentViewController_ExampleTests' do
   pod 'ConsentViewController', :path => '../'
   pod 'Quick'
-  pod 'Nimble'
+  pod 'Nimble', '9.1.0'
```
Comment (andresilveirah) [inline Example/Podfile:18]: Why did you have to pin point the version to `9.1.0`?

File: ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift (hunk 1 above)
Comment (Nevazhnovu) [reply ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:201]: It's the reason of crash in case if "text" key is not present in json

File: Example/Podfile (hunk 2 above)
Comment (Nevazhnovu) [reply Example/Podfile:18]: Please note that this branch was originally made by Vilas

File: ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift (hunk 1 above)
Comment (Nevazhnovu) [reply ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift:201]: @andresilveirah okay this change is not actual since we do have container.decodeIfPresent in develop branch

File: Example/Podfile (hunk 2 above)
Comment (Nevazhnovu) [reply Example/Podfile:18]: @andresilveirah fixed by commit below. Pls review

File: ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift (hunk 1 above)
Comment (Nevazhnovu) [reply ConsentViewController/Classes/SourcePointClient/SPNativeUI.swift]: @andresilveirah this change is reversed

File: Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample-frameworks.sh [hunk 3]
```diff
@@ -113,7 +113,6 @@ install_dsym() {
       rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${DERIVED_FILES_DIR}/${basename}.dSYM" "${DWARF_DSYM_FOLDER_PATH}"
     else
       # The dSYM was not stripped at all, in this case touch a fake folder so the input/output paths from Xcode do not reexecute this script because the file is missing.
-      mkdir -p "${DWARF_DSYM_FOLDER_PATH}"
```
Comment (Nevazhnovu) [inline Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample-frameworks.sh:116]: May it be related to the not passing `NativeMessageUITests` suite? I've tried revert this & no luck as result

File: Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.debug.xcconfig [hunk 4]
```diff
@@ -1,11 +1,10 @@
...
-HEADER_SEARCH_PATHS = $(inherited) "${PODS_CONFIGURATION_BUILD_DIR}/ConsentViewController-iOS/ConsentViewController.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Wormholy/Wormholy.framework/Headers"
-LD_RUNPATH_SEARCH_PATHS = $(inherited) /usr/lib/swift '@executable_path/Frameworks' '@loader_path/Frameworks'
-LIBRARY_SEARCH_PATHS = $(inherited) "${DT_TOOLCHAIN_DIR}/usr/lib/swift/${PLATFORM_NAME}" /usr/lib/swift
-OTHER_LDFLAGS = $(inherited) -framework "ConsentViewController" -framework "Foundation" -framework "Wormholy"
+HEADER_SEARCH_PATHS = $(inherited) "${PODS_CONFIGURATION_BUILD_DIR}/ConsentViewController-iOS/ConsentViewController.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Nimble-iOS/Nimble.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Quick-iOS/Quick.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Wormholy/Wormholy.framework/Headers"
+LD_RUNPATH_SEARCH_PATHS = $(inherited) '@executable_path/Frameworks' '@loader_path/Frameworks'
+OTHER_LDFLAGS = $(inherited) -framework "ConsentViewController" -framework "Foundation" -framework "Nimble" -framework "Quick" -framework "Wormholy" -framework "XCTest" -weak_framework "XCTest"
```
Comment (Nevazhnovu) [inline Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.debug.xcconfig:8]: May it be related to the not passing `NativeMessageUITests` suite? I've tried revert back to `LD_RUNPATH_SEARCH_PATHS = $(inherited) /usr/lib/swift` & no luck as result

File: Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.release.xcconfig [hunk 5]
```diff
@@ -1,11 +1,10 @@
...
-HEADER_SEARCH_PATHS = $(inherited) "${PODS_CONFIGURATION_BUILD_DIR}/ConsentViewController-iOS/ConsentViewController.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Wormholy/Wormholy.framework/Headers"
-LD_RUNPATH_SEARCH_PATHS = $(inherited) /usr/lib/swift '@executable_path/Frameworks' '@loader_path/Frameworks'
-LIBRARY_SEARCH_PATHS = $(inherited) "${DT_TOOLCHAIN_DIR}/usr/lib/swift/${PLATFORM_NAME}" /usr/lib/swift
-OTHER_LDFLAGS = $(inherited) -framework "ConsentViewController" -framework "Foundation" -framework "Wormholy"
+HEADER_SEARCH_PATHS = $(inherited) "${PODS_CONFIGURATION_BUILD_DIR}/ConsentViewController-iOS/ConsentViewController.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Nimble-iOS/Nimble.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Quick-iOS/Quick.framework/Headers" "${PODS_CONFIGURATION_BUILD_DIR}/Wormholy/Wormholy.framework/Headers"
+LD_RUNPATH_SEARCH_PATHS = $(inherited) '@executable_path/Frameworks' '@loader_path/Frameworks'
+OTHER_LDFLAGS = $(inherited) -framework "ConsentViewController" -framework "Foundation" -framework "Nimble" -framework "Quick" -framework "Wormholy" -framework "XCTest" -weak_framework "XCTest"
```
Comment (Nevazhnovu) [inline Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.release.xcconfig:8]: May it be related to the not passing `NativeMessageUITests` suite? I've tried revert back to `LD_RUNPATH_SEARCH_PATHS = $(inherited) /usr/lib/swift` & no luck as result

File: Example/Pods/Target Support Files/Pods-NativeMessageExampleUITests/Pods-NativeMessageExampleUITests-frameworks.sh [hunk 6]
```diff
@@ -113,7 +113,6 @@ install_dsym() {
       rsync --delete -av "${RSYNC_PROTECT_TMP_FILES[@]}" --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "${DERIVED_FILES_DIR}/${basename}.dSYM" "${DWARF_DSYM_FOLDER_PATH}"
     else
       # The dSYM was not stripped at all, in this case touch a fake folder so the input/output paths from Xcode do not reexecute this script because the file is missing.
-      mkdir -p "${DWARF_DSYM_FOLDER_PATH}"
```
Comment (Nevazhnovu) [inline Example/Pods/Target Support Files/Pods-NativeMessageExampleUITests/Pods-NativeMessageExampleUITests-frameworks.sh:116]: May it be related to the not passing `NativeMessageUITests` suite? I've tried revert it & no luck as result

File: Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.debug.xcconfig (hunk 4 above)
Comment (Nevazhnovu) [reply Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.debug.xcconfig:8]: Adding `LIBRARY_SEARCH_PATHS = $(inherited) "${DT_TOOLCHAIN_DIR}/usr/lib/swift/${PLATFORM_NAME}" /usr/lib/swift` is not working either

File: Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.release.xcconfig (hunk 5 above)
Comment (Nevazhnovu) [reply Example/Pods/Target Support Files/Pods-NativeMessageExample/Pods-NativeMessageExample.release.xcconfig:8]: Adding `LIBRARY_SEARCH_PATHS = $(inherited) "${DT_TOOLCHAIN_DIR}/usr/lib/swift/${PLATFORM_NAME}" /usr/lib/swift` is not working either
